pkg/server/upload/storeprovider.go

```diff
@@ -0,0 +1,93 @@
+// Copyright 2025 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package upload
+
+import (
+	"io"
+	"os"
+	"path/filepath"
+)
+
+type UploadFileStoreProvider interface {
+	// Generate a UploadToken for frontend
+	GetUploadToken(id string) UploadToken
+	// Read returns the io.ReadCloser interface to read the file with the given ID.
+	// The caller MUST close the returned ReadCloser.
+	Read(token UploadToken) (io.ReadCloser, error)
+}
+
+type DirectWritableUploadFileStoreProvider interface {
+	// Write writes file with given io.Writer interaface to the file with the given ID.
+	Write(token UploadToken, reader io.Reader) error
+}
+
+// LocalUploadFileStoreProvider is an implementation of UploadFileStore that stores files
+// in the local file system.
+type LocalUploadFileStoreProvider struct {
+	// directoryPath is the folder name where uploaded files are stored.
+	directoryPath string
+}
+
+// NewLocalUploadFileStoreProvider creates a new LocalUploadFileStore.
+// It takes the directory path as an argument and creates the directory
+// if it does not exist.  It returns an error if the directory
+// cannot be created (and does not already exist).
+func NewLocalUploadFileStoreProvider(directoryPath string) (*LocalUploadFileStoreProvider, error) {
+	// Create the directory (and any parent directories) if it doesn't exist.
+	// os.MkdirAll will not return an error if the directory already exists.
+	err := os.MkdirAll(directoryPath, 0755)
+	if err != nil {
+		return nil, err // Return the error if directory creation fails.
+	}
+	return &LocalUploadFileStoreProvider{directoryPath: directoryPath}, nil
```

Comment:
I think a constructor has side-effects is not really great in general. Can we move this to Write?

Reply:
Done